job-runner/chain/redis_repo.go

```diff
@@ -0,0 +1,185 @@
+// Copyright 2017, Square, Inc.
+
+package chain
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/garyburd/redigo/redis"
+)
+
+// ChainKey is the keyspace for serialized Chains
+const CHAIN_KEY = "ChainById"
+
+var (
+	ErrKeyExists           = errors.New("key already exists in redis")
+	ErrKeyNotFound         = errors.New("key not found in redis")
+	ErrMultipleKeysDeleted = errors.New("multiple keys deleted from redis")
+)
+
+// A RedisConnectionPool is used for pooling redis connections.
+type RedisConnectionPool interface {
```

Comment:
Right now this is only used for mocking out a pool in tests. redigo doesn't define this interface, so we have to